Libplanet/Misc/Either/IEither.cs

```diff
@@ -0,0 +1,17 @@
+#nullable enable
+
+using System;
+
+namespace Libplanet.Misc.Either
+{
+    internal interface IEither<TLeft, TRight>
+    {
+        TLeft Left { get; }
+
+        TRight Right { get; }
+
+        TLeft GetLeftOrElse(Func<TRight, TLeft> f);
+
+        TRight GetRightOrElse(Func<TLeft, TRight> f);
```

Comment:
I kind of have a difficult time trying to understand what these methods do. Is it meant to return TLeft or TRight no matter if it's EitherLeft or EitherRight?

Reply:
Originally, I want to use some function like [`std::result::Result<T>::unwrap_or_else()`](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap_or_else).

> Is it meant to return TLeft or TRight no matter if it's EitherLeft or EitherRight?

Maybe it's right. I expect it to support returning `TLeft` or `TRight` with handlers.

Reply:
Ah, now I see where the name comes from. If there's something like `Either.GetLeftOrElse(CalculateLeft)`, it should read as `Get Left from Either or else CalculateLeft(Either.Right)`, right? I kind of still feel like the naming is a bit unintuitive, but I'll leave the other reviewers to consider.

Reply:
Also, are `Left` and `Right` meant to be called as-is? if it's not, I feel like it's not really needed to be exposed in the interface.

Reply:
>  If there's something like Either.GetLeftOrElse(CalculateLeft), it should read as Get Left from Either or else CalculateLeft(Either.Right), right?

You're right.

> Also, are Left and Right meant to be called as-is? if it's not, I feel like it's not really needed to be exposed in the interface.

I added them because other *generic* libraries did but it is Libpalnet's internal implementation so I'll remove them. Thanks for your opinion 🙏🏻